packages/flutter_tools/test/general.shard/flutter_manifest_test.dart

```diff
@@ -82,6 +82,42 @@ flutter:
     expect(flutterManifest.usesMaterialDesign, true);
   });
 
+  testWithoutContext('FlutterManifest knows if generate is provided', () async {
+    const String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  generate: true
+''';
+    final BufferLogger logger = BufferLogger.test();
+    final FlutterManifest flutterManifest = FlutterManifest.createFromString(
+      manifest,
+      logger: logger,
+    );
+
+    expect(flutterManifest.generateSyntheticPackage, true);
+  });
+
+  testWithoutContext('FlutterManifest knows if generate is disabled', () async {
+    const String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  generate: false
```

Comment:
Maybe also test a malformed 'generate: ' to make sure the tool doesn't crash.

Reply:
done